api/metrics/label_gatherer_test.go

```diff
@@ -214,4 +230,64 @@ func TestLabelGatherer_Register(t *testing.T) {
 			require.Equal(test.expectedLabelGatherer, test.labelGatherer)
 		})
 	}
+
+	deregisterTests := []struct {
+		name                  string
+		labelGatherer         *labelGatherer
+		labelValue            string
+		expectedRemoved       bool
+		expectedLabelGatherer *labelGatherer
+	}{
+		{
+			name:                  "remove from nothing",
+			labelGatherer:         &labelGatherer{},
+			labelValue:            firstName,
+			expectedRemoved:       false,
+			expectedLabelGatherer: &labelGatherer{},
+		},
+		{
+			name:                  "remove unknown name",
+			labelGatherer:         firstLabelGatherer(),
+			labelValue:            secondName,
+			expectedRemoved:       false,
+			expectedLabelGatherer: firstLabelGatherer(),
+		},
+		{
+			name:            "remove first name",
+			labelGatherer:   firstLabelGatherer(),
+			labelValue:      firstName,
+			expectedRemoved: true,
+			expectedLabelGatherer: &labelGatherer{
+				multiGatherer: multiGatherer{
+					// We must populate with empty slices rather than nil slices
+					// to pass the equality check.
+					names:     []string{},
+					gatherers: prometheus.Gatherers{},
+				},
+			},
+		},
+		{
+			name:                  "remove second name",
+			labelGatherer:         secondLabelGatherer(),
+			labelValue:            secondName,
+			expectedRemoved:       true,
+			expectedLabelGatherer: firstLabelGatherer(),
+		},
+		{
+			name:                  "remove only first name",
+			labelGatherer:         secondLabelGatherer(),
+			labelValue:            firstName,
+			expectedRemoved:       true,
+			expectedLabelGatherer: onlySecondLabeledGatherer,
+		},
+	}
+	for _, test := range deregisterTests {
+		t.Run(test.name, func(t *testing.T) {
+			require := require.New(t)
+
+			removed := test.labelGatherer.Deregister(test.labelValue)
+			require.Equal(test.expectedRemoved, removed)
+			require.Equal(test.expectedLabelGatherer, test.labelGatherer)
```

Comment:
`require.Equal(test.expectedLabelGatherer, test.labelGatherer)` ?

Reply:
Sorry I don't understand the question